parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReaders.java

```diff
@@ -850,4 +919,42 @@ private TripleIterator<?> firstNonNullColumn(List<TripleIterator<?>> columns) {
       return NullReader.NULL_COLUMN;
     }
   }
+
+  private static class RecordReader<T extends StructLike> extends StructReader<T, T> {
```

Comment:
This returns `Record` so I don't think it needed to be modified. It doesn't return any other subclass of `StructLike`.

Reply:
Internal readers expects `ParquetValueReader<StructLike>` and generic reader expects `ParquetValueReader<Record>`, even though the `Record` extends `StructLike`, because of java generics, we need to return the specific type here. We cannot cast `ParquetValueReader<Record>` to `ParquetValueReader<StructLike>`. 

Hence, this was refactored. Similar to the writer refactoring.